src/sys/mman.rs

```diff
@@ -258,6 +258,33 @@ pub unsafe fn madvise(addr: *mut c_void, length: size_t, advise: MmapAdvise) ->
     Errno::result(libc::madvise(addr, length, advise as i32)).map(drop)
 }
 
+/// Set protection of memory mapping.
+///
+/// See [`mprotect(3)`](http://pubs.opengroup.org/onlinepubs/9699919799/functions/mprotect.html) for
+/// details.
+///
+/// # Safety Calls to `mprotect` are inherently unsafe, as changes to memory protections can lead to
```

Comment:
You need a newline after "Safety".  Also, is there any advice you can offer on how the user can use this function safely?

Reply:
Newline fixed; Emacs got overly enthusiastic about reflowing that paragraph.

As for advice, anything I write would be just rewriting the linked manpage. It's such a broad API with so many potential implications that I think it would serve folks best to just use that as a resource.